drgn_tools/sys.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to view sysinfo data
+"""
+import argparse
+import datetime
+import time
+from typing import Any
+from typing import Dict
+
+from drgn import Architecture
+from drgn import Program
+from drgn import ProgramFlags
+from drgn.helpers.linux import for_each_task
+from drgn.helpers.linux.sched import loadavg
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.cpuinfo import aarch64_get_cpu_info
+from drgn_tools.cpuinfo import x86_get_cpu_info
+from drgn_tools.mm import totalram_pages
+from drgn_tools.table import print_dictionary
+from drgn_tools.util import human_bytes
+
+
+def load_average(prog: Program) -> str:
+    """
+    Return system load averaged over 1, 5 and 15 minutes as a string.
+    """
+    load_avg = loadavg(prog)
+    return "{:0.2f}  ,  {:0.2f} ,  {:0.2f}".format(
+        load_avg[0],
+        load_avg[1],
+        load_avg[2],
+    )
+
+
+def get_mem(prog: Program) -> str:
+    """
+    Returns the total memory of the system, as a human-readable string
+    """
+    total_pages = totalram_pages(prog)
+    pagesize = int(prog["PAGE_SIZE"])
+    return human_bytes(total_pages * pagesize)
+
+
+def task_info(prog: Program) -> str:
+    """
+    Returns the task info of system as a string
+    """
+    task_running = 0
+    task_int = 0
+    task_unint = 0
+    tasks_count = prog["nr_cpu_ids"]
+    for task in for_each_task(prog):
+        tasks_count += 1
+        try:
+            if prog.type("struct task_struct").has_member("__state"):
+                task_state = task.__state
+            else:
+                task_state = task.state
+        except LookupError:
+            task_state = task.__state
+        if task_state == 0:
+            task_running += 1
+        elif task_state == 1:
+            task_int += 1
+        elif task_state == 2:
+            task_unint += 1
+    return f"{int(tasks_count)} [R:{task_running} D:{task_unint} S:{task_int}]"
```

Comment:
* Drgn conveniently has a helper to convert the states to character codes: [`task_state_to_char()`](https://drgn.readthedocs.io/en/latest/helpers.html#drgn.helpers.linux.sched.task_state_to_char). You can use this so that you don't need to hard-code anything.
* You seem to have over-complicated your `task.__state` logic, you can use an if statement, or a try/except, but you don't need both. But in any case, with the above helper, you don't need that either.
* I'm a bit confused why `tasks_count` starts at `prog["nr_cpu_ids"]` and not 0?
```suggestion
    state_count = {}
    tasks_count = 0
    for task in for_each_task(prog):
        tasks_count += 1
        state_code = task_state_to_char(task)
        state_count[state_code] = state_count.get(state_code, 0) + 1
    # ensure the first categories are R(unning), D (uninterruptible), S (interruptible)
    output = f"{tasks_count}"
    for code in ("R", "D", "S"):
        output += " {code}:{state_count.get(code, 0)}"
        state_count.pop(code)
    # the remaining items in any order
    for code, count in state_count.items():
        output += " {code}:{count}"
    return output
```

The benefit here being that you get other state codes too: e.g. Z, I, X